scipy/optimize/_direct_py.py

```diff
@@ -212,16 +212,12 @@ def direct(
     ub = np.ascontiguousarray(bounds.ub, dtype=np.float64)
 
     # validate bounds
+    # check that lower bounds are smaller than upper bounds
     if not np.all(lb < ub):
         raise ValueError('Bounds are not consistent min < max')
-    if not len(lb) == len(ub):
-        raise ValueError('Bounds do not have the same dimensions')
-
-    # check for infs and nans
+    # check for infs
     if (np.any(np.isinf(lb)) or np.any(np.isinf(ub))):
         raise ValueError("Bounds must not be inf.")
-    if (np.any(np.isnan(lb)) or np.any(np.isnan(ub))):
-        raise ValueError("Bounds must not be NaN.")
```

Comment:
This is covered by the test above that `lb < ub`. For NaN it throws an error that comparison is not possible, so this explicit check for NaNs is never reached.

Reply:
I don't think it will raise an error, but if an element of either array is NaN, then the corresponding comparison will always return False, so you'll get the error that Bounds are not consistent.